src/constants.lua

```diff
@@ -36,70 +42,122 @@ function constants.hoursToMs(hours)
 	return hours * constants.minutesToMs(60)
 end
 
+-- TOKEN SUPPLY
+constants.TOTAL_TOKEN_SUPPLY = constants.ARIOToMARIO(1000000000) -- 1 billion tokens
+constants.DEFAULT_PROTOCOL_BALANCE = constants.ARIOToMARIO(50000000) -- 50M ARIO (TODO: may change to 65M on mainnet)
+constants.MIN_UNSAFE_ADDRESS_LENGTH = 1
+constants.MAX_UNSAFE_ADDRESS_LENGTH = 128
+
 -- EPOCHS
-constants.defaultEpochDurationMs = constants.daysToMs(1)
-constants.distributionDelayMs = constants.minutesToMs(40) -- 40 minutes
-constants.maximumRewardRate = 0.001
-constants.minimumRewardRate = 0.0005
-constants.rewardDecayStartEpoch = 365
-constants.rewardDecayLastEpoch = 547
-constants.observerRewardRatio = 0.1
-constants.gatewayOperatorRewardRatio = 0.9
+constants.DEFAULT_EPOCH_SETTINGS = {
+	prescribedNameCount = 2, -- observers choose 8 names per epoch
+	maxObservers = 50,
+	epochZeroStartTimestamp = 1719900000000, -- July 9th,2024 00:00:00 UTC (TODO: set this on mainnet process, likely two weeks after TGE)
+	durationMs = constants.daysToMs(1), -- 1 day
+}
+
+-- DISTRIBUTIONS
+--[[
+	Distribution rewards will be 0.1% of the protocol balance for the first year,
+	then decay linearly to 0.05% of the protocol balance after the first year until 1.5 years.
+	After 1.5 years, the reward rate will be 0.05% of the protocol balance.
+
+	TODO: ensure this is enabled on mainnet
```

Comment:
Note to self: check for tests for this algorithm.

Reply:
validated and updated tests in [13ce11e](https://github.com/ar-io/ar-io-network-process/pull/363/commits/13ce11e368e0a11f95a70792edcb18f348651ef9)

Reply:
This is the commit that restores deepCopy of the gateway returned by leaveNetwork 🤔

Reply:
I see the test in the other commits.